MaintenanceManager/MaintenanceManager.cpp

```diff
@@ -216,67 +217,234 @@ namespace WPEFramework {
          }
 
         void MaintenanceManager::task_execution_thread(){
-            int task_count=3;
-            int i=0;
+            uint8_t i=0;
             string cmd="";
+            bool internetConnectStatus=false;
 
-            LOGINFO("Reboot_Pending :%s",g_is_reboot_pending.c_str());
+            /* cleanup if not empty */
+            if(!tasks.empty()){
+                tasks.erase (tasks.begin(),tasks.end());
+            }
 
-            bool internetConnectStatus = isDeviceOnline();
+            /* Controlled by CFLAGS */
+#if defined(SUPPRESS_MAINTENANCE)
+            bool activationStatus=false;
+            bool skipFirmwareCheck=false;
 
-            MaintenanceManager::_instance->onMaintenanceStatusChange(MAINTENANCE_STARTED);
-            /*  In an unsolicited maintenance we make sure only after
-             *  after DCM task activities are started.
-             */
+            /* Activation check */
+            activationStatus = isDeviceActivated(skipFirmwareCheck);
 
-            /* we add the task in a loop */
+            /* we proceed with network check only if
+             * "activation-connect", "activation-ready"
+             * "not-activated", "activated" */
+            if(activationStatus){
+                /* Network check */
+                internetConnectStatus = isDeviceOnline();
+            }
+#else
+            internetConnectStatus = isDeviceOnline();
+#endif
+            LOGINFO("Reboot_Pending :%s",g_is_reboot_pending.c_str());
+
+            MaintenanceManager::_instance->onMaintenanceStatusChange(MAINTENANCE_STARTED);
+#if defined(SUPPRESS_MAINTENANCE)
```

Comment:
Pls remove this SUPPRESS_MAINTENANCE build flag to simplify the code. If we cannot read activation status from authservice then we can default to existing behavior.

Reply:
This is specific to Platco & Hisense, so we are restricting this to these platforms based on the distro feature. Hence activation check is only for the aforesaid platforms, rest of the platforms we don't enable it.